### PR TITLE
test for overlapping symbols

### PR DESCRIPTION
**Pull request by EtDu**
Added a test to ensure no symbols overlap. Ref: #451

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -88,3 +88,22 @@ test('only permitted fields should be used', function (t) {
 
   t.end()
 })
+
+test('symbols should not overlap', function (t) {
+  const symbols = Object.keys(contractMap).map(contract => contractMap[contract].symbol)
```
Comment (kumavis) [inline test/index.js]: Use Object.values here

File: test/index.js [hunk 2]
```diff
@@ -88,3 +88,22 @@ test('only permitted fields should be used', function (t) {
...
+      const symbol = symbols[i]
+      
+      if (symbol in symbolsCheck && symbol !== undefined) {
+        return symbol
+      }
+      symbolsCheck[symbol] = true
+    }
+  })()
+  
+  t.notOk(duplicateSymbol, "Symbols should not overlap")
```
Comment (kumavis) [inline test/index.js]: Include the overlapping name in quotes in the error message here

File: test/index.js [hunk 3]
```diff
@@ -88,3 +88,22 @@ test('only permitted fields should be used', function (t) {
 
   t.end()
 })
+
+test('symbols should not overlap', function (t) {
+  const symbols = Object.keys(contractMap).map(contract => contractMap[contract].symbol)
+  const symbolsCheck = Object.create(null)
+
+  const duplicateSymbol = (() => {
+    for (var i = 0; i < symbols.length; ++i) {
```
Comment (kumavis) [inline test/index.js]: Should be able to do `for (const symbol of symbols)`

File: test/index.js [hunk 4]
```diff
@@ -88,3 +88,22 @@ test('only permitted fields should be used', function (t) {
 
   t.end()
 })
+
+test('symbols should not overlap', function (t) {
+  const symbols = Object.keys(contractMap).map(contract => contractMap[contract].symbol)
+  const symbolsCheck = Object.create(null)
```
Comment (kumavis) [inline test/index.js]: Better to use a `Map` or `Set` than an object. Will need to change the `in` operator to match

**Comment by kumavis** [comment]
dank

**Comment by Dmitr33** [comment]
[]()@EtDu

**Pull request merged**